lib/styleguide.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const _ = require('lodash'),
+  clayUtils = require('clayutils'),
+  files = require('nymag-fs'),
+  path = require('path');
+
+/**
+ * Looks for CSS files associated with a component's variation
+ *
+ * @param {String} styleguide - name of the styleguide that is set in the sites config
+ * @returns {Array} variations
+ */
+function findVariations(styleguide) {
+  const stylePath = `styleguides/${styleguide}/components`,
+    stylesheets = files.getFiles(stylePath),
+    variations = _.filter(stylesheets, function (file) {
+      // checks for underscores which denotes variations and make sure it's a
+      // css file
+      return file.includes('_') && file.includes('.css');
+    });
+
+  return variations;
+}
+
+/**
+* Grabs variations of all components and returns an array organized by components
+*
+* @param {String} styleguide - name of the styleguide that is set in the sites' config
+* @returns {Array} componentVariations - an array of objects that corresponds to
+* a component's name and then its variations
+*/
+function getVariations(styleguide) {
+  const foundVariations = styleguide ? findVariations(styleguide) : [],
+    componentVariations = {};
+
+  if (foundVariations.length !== 0) {
+    _.forEach(foundVariations, function (variant) {
+      let component = variant.split('_')[0],
+        variantName = path.basename(variant, '.css');
+
+      (componentVariations[component] || (componentVariations[component] = [])).push(variantName);
+    });
+  }
+  return componentVariations;
+}
+
+/**
+* Loops through an array of layout data and sets a default for components that
+* have variations but don't have any set. Note: this mutates the array!
+*
+* @param {Array} array
+* @param {object} variations - all components with variations, usually generated by getVariations
+*/
+function setDefaultVariation(array, variations) {
+  _.forEach(array, function (val) {
+    const componentName = _.isObject(val) && val._ref ? clayUtils.getComponentName(val._ref) : undefined;
+
+    if (componentName && variations[componentName]) {
+      if (!val.componentVariation) {
+        _.set(val, 'componentVariation', componentName);
+      }
+    }
+
+    if (val.content) {
```

Comment:
Does this assume that a component can only have a component list at its `content` property?

Reply:
yes, is this a bad assumption? Would it be better to check if the property is an array?